caffe2/CMakeLists.txt

```diff
@@ -702,7 +703,19 @@ if(USE_ROCM)
   endif()
 elseif(USE_CUDA)
   set(CUDA_LINK_LIBRARIES_KEYWORD PRIVATE)
-  cuda_add_library(torch_cuda ${Caffe2_GPU_SRCS})
+  if(CUDA_SEPARABLE_COMPILATION)
+    # Separate compilation fails when kernels using `thrust::sort_by_key`
```

Comment:
Nit: any external links to posts, issues, etc. you'd want to add here? Just might be something we'll all forget about in a years time or so.

Reply:
There are no external posts about this particular issue, but there are several issues reports against thrust/cub, for example https://github.com/thrust/thrust/issues/928

Reply:
Just throwing that link in there would be useful even though it's on a different repo.